src/pumps/batch.rs

```diff
@@ -0,0 +1,73 @@
+use tokio::{
+    sync::mpsc::{self, Receiver},
+    task::JoinHandle,
+};
+
+use crate::Pump;
+
+pub struct BatchPump<T> {
+    pub(crate) n: usize,
+    batch: Vec<T>,
+}
+
+impl<T> BatchPump<T> {
+    pub fn new(n: usize) -> Self {
+        Self {
+            n,
+            batch: Vec::with_capacity(n),
```

Comment:
the batch vec can be local to the spawn method, no reason to include it in the struct